cookbooks/bach_spark/recipes/default.rb

```diff
@@ -11,6 +12,35 @@
   mode 0755
 end
 
+template "#{spark_bin_dir}/conf/spark-defaults.conf" do
+  source "spark-defaults.conf.erb"
+  mode 0755
+end
+
+bash "create-hdfs-spark-history-dir" do
+  code <<-EOH
+  hdfs dfs -mkdir -p #{hdfs_url}/spark-history/
+  hdfs dfs -chmod -R 1777 #{hdfs_url}/spark-history/
+  hdfs dfs -chown -R hdfs:hdfs #{hdfs_url}/spark-history
+  EOH
+  user "hdfs"
+  not_if "sudo -u hdfs hadoop dfs -test -d #{hdfs_url}/spark-history"
```

Comment:
You do need to run the guard attribute as hdfs but that's now been done to remove the sudo's/